Migration material components 1.1.0

mlcsouza opened the pull request:
In this PR, the Android Support Design Library was replaced by Material Components.
All components were migrated to AndroidX correspondence

pmunizferreira commented on build.gradle:
```diff
@@ -47,4 +47,5 @@ ext {
     androidxJunitVersion = "1.1.1"
     truthVersion = '1.2.0'
     robolectricVersion = '4.3'
+    material = '1.1.0'
```
que tal mantermos o padrao e renomearmos para `materialVersion`? :/

pmunizferreira replied:
`supportLibraryVersion` ainda esta em uso?

pmunizferreira commented on designsystem/src/main/kotlin/com/natura/android/button/DsPrimaryButton.kt:
```diff
@@ -2,11 +2,11 @@ package com.natura.android.button
 import android.view.View
+import androidx.appcompat.widget.AppCompatTextView
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.core.content.ContextCompat
```
existe um card pra removermos esse componente. Ele nao esta em uso e foi adicionado aqui sem revisao. Lucas solicitou sua remocao.
Alem disso podemos usar o nativo do android.

viniciusponciano-tw commented on designsystem/src/main/res/layout/ds_default_error.xml:
```diff
@@ -39,22 +40,22 @@
+        android:layout_marginStart="16dp"
+        android:layout_marginEnd="16dp"
         android:layout_marginBottom="@dimen/ds_spacing_little"
+        android:text="RECARERGAR"
```
A label está incorreta o certo seria RECARREGAR

mlcsouza replied:
Bem observado!

mlcsouza commented:
# Description
- Update on core components lib from Android Support Design V28 to Material Android Components from Android V1.1.0
- We added more than 70 new color references in the lib
- We added Natura, Avon and The Body Shop color themes in light and dark mode

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
